drud/ddev
# Add TYPO3 11 Backend URL rewriting support and php7.4 default for TYPO3

## The Problem/Issue/Bug:

With the [patch 67083](https://review.typo3.org/c/Packages/TYPO3.CMS/+/67083) TYPO3 introduces URL rewriting for the Backend and therefor the Nginx configuration needs to be adapted like explained for Apache see https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/master/Breaking-93048-BackendURLRewrites.html#migration. Currently after the login to the backend this error occurs:
`The page did not exist or was inaccessible. Reason: The requested page does not exist`

Also minimal support for PHP was raised to 7.4 which must also be addressed.

## How this PR Solves The Problem:

This change adds an additional configuration for Nginx and the default PHP version for the project type typo3 was changed to 7.4.

## Manual Testing Instructions:

Without this change the backend will fail after login with this error: `The page did not exist or was inaccessible. Reason: The requested page does not exist`

Testing instructions:

```
ddev config --project-type=typo3 --docroot=public --create-docroot
ddev composer create "typo3/cms-base-distribution:dev-master"
ddev exec touch public/FIRST_INSTALL
ddev launch
```

- Finish the Install Wizard
- Login to the Backend

If no error is shown the test is successful.

This change was tested with TYPO3 11 (master), 10.4 and 9.5 and works fine so far.

## Automated Testing Overview:


## Related Issue Link(s):

- https://github.com/TYPO3/TYPO3.CMS/commit/07bfbc849140d6164e9a146b70ab3643d7557ef8
- https://github.com/TYPO3/TYPO3.CMS.BaseDistribution/pull/30

## Release/Deployment notes:
